sql-plugin/src/main/scala/org/apache/spark/sql/rapids/GpuJsonToStructs.scala

```diff
@@ -33,12 +33,37 @@ case class GpuJsonToStructs(
     timeZoneId: Option[String] = None)
     extends GpuUnaryExpression with TimeZoneAwareExpression with ExpectsInputTypes
         with NullIntolerant {
+
+  // Construct a dummy json string given an input schema, for example:
+  // schema = StructType([StructField("a", StringType), StructField("b", IntegerType)])
+  // returns "{"a": "", "b": 0}"
+  private def constructEmptyRow(): String = {
+    schema match {
+      case struct: StructType if (struct.fields.length > 0) => {
+        val res = struct.fields.foldRight ("") ((field, acc) => 
+          field.dataType match {
+            case IntegerType => "\"" + field.name + "\": 0, " + acc
+            case _ => "\"" + field.name + "\": \"\", " + acc
+          }
+        )
+        "{" + res.dropRight(2) + "}"
+      }
+      case _ => "{}"
+    }
```

Comment:
could simplify with first creating an array of kv pairs and then mkString it
```suggestion
    val jsonFields: Array[String] = schema match {
      case StructType(fields) => {
        fields.map {
          case StructField(name, IntegerType, _, _) => s""" "${name}" : 0 """
          case StructField(name, _, _, _)           => s""" "${name}" : "" """
        }
      }
      case _ => Array.empty[String]
    }
    jsonFields.mkString("{", ",", "}")
```

Reply:
After further testing, I found we need to make this function recursive to take care of nested `StructType`. I will see if I can apply the above logic there. Thank you!

Reply:
I think you should be able to after making schema the function param, you can recurse by adding the 3td case around L45